pkg/server/grpc.go

```diff
@@ -128,3 +133,41 @@ func loadTLSCredentials(certFile, keyFile string) (credentials.TransportCredenti
 
 	return credentials.NewTLS(tlsConfig), nil
 }
+
+func customMetricsInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	start := time.Now()
+
+	resp, err := handler(ctx, req)
```

Comment:
I think we need to better indicate that this err should be returned. A name like grpcServiceErr makes more sense to me.